### PR TITLE
fix(angular): fix e2e tests to support pnpm v7

### PR DESCRIPTION
**Pull request by meeroslav**
## Related Issue(s)

Cherry picks https://github.com/nrwl/nx/pull/15695, which should be merged in first

### CODE REVIEW

File: e2e/utils/command-utils.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ export function runCommand(
 ): string {
   const { failOnError, ...childProcessOptions } = options ?? {};
   try {
-    const r = execSync(`${command}${isVerbose() ? ' --verbose' : ''}`, {
+    const r = execSync(command, {
```
Comment (meeroslav) [inline e2e/utils/command-utils.ts:73]: This is fixed in https://github.com/nrwl/nx/pull/15541

File: e2e/utils/command-utils.ts [hunk 2]
```diff
@@ -279,10 +279,7 @@ export function runNgAdd(
   try {
     const pmc = getPackageManagerCommand();
     packageInstall(packageName, undefined, version);
-    const fullCommand = pmc.run(
-      `ng g ${packageName}:ng-add`,
-      `${command}${isVerbose() ? ' --verbose' : ''}` ?? ''
-    );
+    const fullCommand = pmc.run(`ng g ${packageName}:ng-add`, command ?? '');
```
Comment (meeroslav) [inline e2e/utils/command-utils.ts:282]: This is fixed in https://github.com/nrwl/nx/pull/15541

File: e2e/utils/command-utils.ts [hunk 3]
```diff
@@ -369,9 +366,7 @@ export function runLernaCLI(
 ): string {
   try {
     const pm = getPackageManagerCommand();
-    const fullCommand = `${pm.runLerna} ${command}${
-      isVerbose() ? ' --verbose' : ''
-    }`;
+    const fullCommand = `${pm.runLerna} ${command}`;
```
Comment (meeroslav) [inline e2e/utils/command-utils.ts:369]: This is fixed in https://github.com/nrwl/nx/pull/15541

File: e2e/angular-core/src/ng-add.test.ts [hunk 4]
```diff
@@ -72,7 +72,7 @@ describe('convert Angular CLI workspace to an Nx workspace', () => {
   });
 
   afterEach(() => {
-    cleanupProject();
+    cleanupProject({ skipReset: true });
```
Comment (meeroslav) [inline e2e/angular-core/src/ng-add.test.ts:75]: These tests do not start with standard create-nx-workspace so running `nx reset` fails after the test.

File: packages/angular/src/generators/host/files/src/main.server.ts__tmpl__ [hunk 5]
```diff
@@ -39,7 +39,7 @@ export function app(): express.Express {
   );
 
   // All regular routes use the Universal engine
-  server.get('*', (req, res) => {
+  server.get('*', (req: any, res: any) => {
```
Comment (meeroslav) [inline packages/angular/src/generators/host/files/src/main.server.ts__tmpl__:42]: Build fails due to Typescript errors

File: packages/angular/src/generators/host/lib/add-ssr.ts [hunk 6]
```diff
@@ -65,7 +67,10 @@ export async function addSsr(tree: Tree, options: Schema, appName: string) {
       express: expressVersion,
       '@module-federation/node': moduleFederationNodeVersion,
     },
-    {}
+    {
+      '@types/cors': typesCorsVersion,
```
Comment (meeroslav) [inline packages/angular/src/generators/host/lib/add-ssr.ts:71]: Build fails due to unknown types

File: package.json [hunk 7]
```diff
@@ -46,7 +46,7 @@
     "@babel/preset-typescript": "^7.15.0",
     "@cypress/react": "^6.0.0",
     "@floating-ui/react": "0.19.2",
-    "@monodon/rust": "1.0.0",
+    "@monodon/rust": "1.1.1",
```
Comment (meeroslav) [inline package.json:49]: This is cherry picked from https://github.com/nrwl/nx/pull/15695

File: yarn.lock [hunk 8]
```diff
@@ -4281,10 +4281,10 @@
     npmlog "^6.0.2"
     write-file-atomic "^4.0.1"
 
-"@ltd/j-toml@1.24.0":
-  version "1.24.0"
-  resolved "https://registry.yarnpkg.com/@ltd/j-toml/-/j-toml-1.24.0.tgz#4c993610586964fb21fd38a84bb41913f6e4f5ff"
-  integrity sha512-XWEvSNLJ2YeCvlNDd/DQfTMldJqBZWX8+RKgPMY2i3MShqsB9XvtY2UfHqcuPb+eRrNO1WR7w4xlNfzCzop+Ig==
+"@ltd/j-toml@1.38.0":
```
Comment (meeroslav) [inline yarn.lock:4284]: This is cherry picked from https://github.com/nrwl/nx/pull/15695

**Review by mandarini — APPROVED** [review]

File: e2e/utils/create-project-utils.ts [hunk 9]
```diff
@@ -81,7 +81,7 @@ export function newProject({
       packageInstall(packages.join(` `), projScope);
 
       // stop the daemon
-      execSync('nx reset', {
+      execSync('npx nx reset', {
```
Comment (barbados-clemens) [inline e2e/utils/create-project-utils.ts:84]: should'd we use the `packageManaer` instead?

Comment (meeroslav) [reply e2e/utils/create-project-utils.ts:84]: We could but it's not necessary in this case. This should happen only once per test project run and ensures daemon is stopped before we copy workspace to another folder.

**Review by Coly010 — APPROVED** [review]

File: packages/angular/src/generators/host/files/src/main.server.ts__tmpl__ (hunk 5 above)
Comment (leosvelperez) [reply packages/angular/src/generators/host/files/src/main.server.ts__tmpl__:42]: These TS type errors are very interesting. They only arise with pnpm v7. I tried with pnpm 6 and it's all good. Also, the IDE doesn't report such an issue, even when the packages are installed with pnpm v7.

That said, by adding `@types/express` the types are inferred correctly and therefore, we don't need to add `any`.

File: e2e/angular-core/src/ng-add.test.ts (hunk 4 above)
Comment (leosvelperez) [reply e2e/angular-core/src/ng-add.test.ts:75]: They don't start with `create-nx-workspace` but still end up being Nx workspaces at the end of the tests. Each test converts the initial Angular CLI workspace into an Nx workspace. Surely that starts the daemon and needs to be reset, isn't it?

I can see a couple of test cases don't run any command, so maybe those don't have the daemon running, but the rest run commands. We might need to inline the `cleanupProject` calls to the end of each test case and pass the `skipReset: true` to those than don't run commands.

**Review by leosvelperez — CHANGES_REQUESTED** [review]
Left a couple of comments in some of the comments you initially added.

File: packages/angular/src/generators/host/files/src/main.server.ts__tmpl__ (hunk 5 above)
Comment (meeroslav) [reply packages/angular/src/generators/host/files/src/main.server.ts__tmpl__]: You are correct, adding types solved it

File: e2e/angular-core/src/ng-add.test.ts (hunk 4 above)
Comment (meeroslav) [reply e2e/angular-core/src/ng-add.test.ts]: Actually, by ensuring proper cleanup the `nx reset` issue is now gone so I reverted in back to clean `cleanupProject`

**Review by leosvelperez — APPROVED** [review]

**Pull request merged**